node/heartbeat/sender.go

```diff
@@ -6,16 +6,17 @@ import (
 
 	"github.com/ElrondNetwork/elrond-go/crypto"
 	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/sharding"
 )
 
 // Sender periodically sends heartbeat messages on a pubsub topic
 type Sender struct {
-	peerMessenger PeerMessenger
-	singleSigner  crypto.SingleSigner
-	privKey       crypto.PrivateKey
-	marshalizer   marshal.Marshalizer
-	topic         string
-	shardID       uint32
+	peerMessenger    PeerMessenger
+	singleSigner     crypto.SingleSigner
+	privKey          crypto.PrivateKey
+	marshalizer      marshal.Marshalizer
+	topic            string
+	shardCoordinator *sharding.Coordinator
```

Comment:
no need for pointer here. Coordinator is an interface

Reply:
done